src/provider/systemdmanager/manager.rs

```diff
@@ -0,0 +1,262 @@
+//* A systemd unit manager that allows managing systemd services
+
+use crate::provider::systemdmanager::service::SystemDUnit;
+use anyhow::anyhow;
+use dbus::blocking::SyncConnection;
+use dbus::Path;
+use log::{debug, error, info};
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
+use std::time::Duration;
+
+pub enum UnitTypes {
+    Service,
+}
+
+pub struct Manager {
+    units_directory: PathBuf,
+    systemd_connection: SystemDConnection,
+}
+
+impl Manager {
+    pub fn new(units_directory: PathBuf, user_mode: bool) -> Self {
+        Manager {
+            units_directory,
+            systemd_connection: SystemDConnection::new(user_mode),
+        }
+    }
+
+    // Write the unit file to disk and enable the service
```

Comment:
I've re-used the name from fleet's implementation of a [systemd manager](https://github.com/coreos/fleet/blob/master/systemd/manager.go#L102) - but am really not attached to it at all.
The systemctl equivalent would be the _link_ command, maybe that?

Reply:
I don't really have a strong opinion.
I thought the equivalent is `enable`? Doesn't that create the symlinks?

Reply:
Enable creates the symlink so that it is started at boot, load creates the symlink to make the service _targetable_ by systemctl start / stop and the like.

I have to admit that I am a bit confused by systemd naming as well, though..

Reply:
Uhmm...okay. I'll leave it up to you :)

Reply:
I have refactored pretty much the entire thing and chose (hopefully) names that speak to what their use is. I mostly decided to use systemd-terms except for create_unit and remove_unit.